ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java

```diff
@@ -50,6 +65,9 @@
   private final String queriesFile;
   private final UdfLoader udfLoader;
   private final CountDownLatch shutdownLatch = new CountDownLatch(1);
+  private final Map<Class<? extends Statement>, BiConsumer<String, Statement>>
```

Comment:
This could be static, immutable and initialised once for the class, e.g. consider:

```
  // Define a suitable 'handler' interface:
  private interface Handler<T extends Statement> {
    void handle(StandaloneExecutor executor, String queryString, T statement);
  }

  // Build your static & immutable map of handler functions:
  private static final Map<Class<? extends Statement>, Handler<Statement>> HANDLERS =
      ImmutableMap.<Class<? extends Statement>, Handler<Statement>>builder()
          .put(SetProperty.class, castHandler(StandaloneExecutor::handleSetProperty, SetProperty.class))
          .put(CreateStream.class, castHandler(StandaloneExecutor::handleCreateStream, CreateStream.class))
      .build();

  // Each member function is passed its required arguments, (pre-cast to the right type).
  @SuppressWarnings("unused")
  private void handleSetProperty(final String ignored, final SetProperty statement) {
    configProperties.put(statement.getPropertyName(), statement.getPropertyValue());
  }
  
  @SuppressWarnings("unused")
  private void handleCreateStream(final String queryString, final CreateStream ignored) {
    ksqlEngine.buildMultipleQueries(queryString, ksqlConfig, configProperties);
  }

  // Define a default handler to call when no other handler registered.
  @SuppressWarnings("unused")
  private void defaultHandler(final String queryString, final Statement ignored) {
    log.warn(String.format(
        "Ignoring statements: %s"
            + "%nOnly DDL (CREATE STREAM/TABLE, DROP STREAM/TABLE, SET, UNSET) "
            + "and DML(CSAS, CTAS and INSERT INTO) statements can run in standalone mode.",
        queryString));
  }

 // A little casting magic in a single place:
  private static <T extends Statement> Handler<Statement> castHandler(final Handler<? super T> handler, final Class<T> type) {
    return (executor, queryString, statement) -> handler.handle(executor, queryString, type.cast(statement));
  }

 // Then executeStatements can simply be:
 private void executeStatements(final String queries) {
    final List<Pair<String, Statement>> statementPairs =
        ksqlEngine.parseStatements(queries, ksqlEngine.getMetaStore().clone());

    for (final Pair<String, Statement> statementPair: statementPairs) {
      final Statement statement = statementPair.getRight();

      HANDLERS
          .getOrDefault(statement.getClass(), StandaloneExecutor::defaultHandler)
          .handle(this, statementPair.getLeft(), statement);
    }
  }
```

The benefit of this, (aside from only having to build the map once), is that each handler is in its own function. Small, targeted functions are good: they are easy to grok and clear and pass check style rules on function complexity to boot.

Reply:
Updated based on your suggestion!